nixos/modules/config/users-groups.nix

```diff
@@ -557,18 +557,77 @@ in {
         # password or an SSH authorized key. Privileged accounts are
         # root and users in the wheel group.
         assertion = !cfg.mutableUsers ->
-          any id (mapAttrsToList (name: cfg:
-            (name == "root"
-             || cfg.group == "wheel"
-             || elem "wheel" cfg.extraGroups)
-            &&
-            ((cfg.hashedPassword != null && cfg.hashedPassword != "!")
-             || cfg.password != null
-             || cfg.passwordFile != null
-             || cfg.openssh.authorizedKeys.keys != []
-             || cfg.openssh.authorizedKeys.keyFiles != [])
-          ) cfg.users);
+          (let
```

Comment:
IIUC, this assertion can not be accurate if `security.sudo.configFile` is overwritten or `security.sudo.extraConfig` is set. I see two options:
- print a warning if that's the case and do not check anything;
- still check things, but add an explicit "breakglass" option to disable this assertion.